### PR TITLE
Kernel/aarch64: Support reading the command line via the RPi Mailbox

### PR DESCRIPTION
**Pull request by BertalanD**
Image showing the kernel command line printed in the debug console

This PR reuses the existing `RPi::Mailbox` interface to read the command line via a VideoCore-specific mailbox message.

It will eventually allow us to run the system in self-test mode by passing the appropriate parameter via the command line.

This will have to be replaced if that interface starts being smarter, as this is needed very early, and nothing guarantees that a smarter Mailbox interface wouldn't need to allocate or log, which is a no-no during early boot. The portable way would be to parse the `/chosen/bootargs` property of the device tree, but we currently lack the scaffolding for doing that.

Support for this in QEMU relies on a patch that has not yet been accepted upstream, but is available via our `Toolchain/BuildQEMU.sh` script. It should, however, work on bare metal.

### CODE REVIEW

File: Kernel/Arch/aarch64/RPi/Mailbox.cpp [hunk 1]
```diff
@@ -130,4 +130,37 @@ u32 Mailbox::query_firmware_version()
...
+{
+    // We want to use the user-provided buffer rather than a fixed-size one on the stack,
+    // so we need to construct the message manually.
+    auto aligned_buffer = buffer.align_to(16);
+    if (aligned_buffer.size() < 24)
+        return ""sv;
+
+    auto max_response_length = aligned_buffer.size() - 24;
+
+    auto* message = reinterpret_cast<u32*>(buffer.data());
```
[the remark this answers is not in the document]
Comment (BertalanD) [reply Kernel/Arch/aarch64/RPi/Mailbox.cpp]: Yes. :yakpalm:

File: Kernel/Arch/aarch64/RPi/Mailbox.cpp [hunk 2]
```diff
@@ -130,4 +130,37 @@ u32 Mailbox::query_firmware_version()
...
+    message[2] = 0x0005'0001; // Query command line
+    message[3] = max_response_length;
+    message[4] = max_response_length;
+
+    message[aligned_buffer.size() / sizeof(u32) - 1] = 0;
+
+    if (!the().send_queue(message, aligned_buffer.size()))
+        return ""sv;
+
+    auto response_length = message[4] & 0x7fff'ffff;
```
Comment (nico) [inline Kernel/Arch/aarch64/RPi/Mailbox.cpp:158]: maybe add a comment why you're masking this

**Review by nico — APPROVED** [review]

File: Kernel/Arch/aarch64/RPi/Mailbox.cpp (hunk 2 above)
Comment (nico) [reply Kernel/Arch/aarch64/RPi/Mailbox.cpp:158]: (ie

Response codes:
b31 set: response
b30-b0: value length in bytes

from https://github.com/raspberrypi/firmware/wiki/Mailbox-property-interface)

File: Kernel/Arch/init.cpp [hunk 3]
```diff
@@ -130,7 +130,7 @@ READONLY_AFTER_INIT PhysicalAddress boot_pdpt;
 READONLY_AFTER_INIT PhysicalAddress boot_pd0;
```
Comment (FireFox317) [inline Kernel/Arch/init.cpp:130]: Typo in commit message: `Kenrnel` -> `Kernel`

**Review by FireFox317 — COMMENTED** [review]
Thanks for the `Tested-By` tag!

**Pull request merged**